Add tutorial for Kubernetes launch cluster (#4403)
DataBiosphere/toil: docs/running/cloud/amazon.rst

command line option to ``toil launch-cluster``.

For example, to launch a Toil cluster with a Kubernetes scheduler, run: ::

    (venv) $ toil launch-cluster <cluster-name> \
            --provisioner=aws \
            --clusterType kubernetes \
            --zone us-west-2a \
            --keyPairName wlgao@ucsc.edu \
            --leaderNodeType t2.medium \
            --leaderStorage 50 \
            --nodeTypes t2.medium -w 1-4 \
            --nodeStorage 20 \
            --logDebug

Behind the scenes, Toil installs kubeadm and configures kubelet on the Toil
leader and all worker nodes. This Toil cluster can then schedule jobs using
Kubernetes.

.. note::
    You should set at least one worker node, otherwise Kubernetes would not be
    able to schedule any jobs. It is also normal for this step to take a while.

Below is a tutorial on how to launch a Toil job on this newly created cluster.
As a demostration, we will use :download:`sort.py <../../../src/toil/test/sort/sort.py>`
again, but run it on a Toil cluster with Kubernetes. First, download this file
and put it to the current working directory.

We then need to copy over the workflow file and SSH into the cluster: ::

>     (venv) $ toil rsync-cluster -z us-west-2a <cluster-name> tutorial_multiplejobs3.py :/root

Reviewer: ```suggestion
    (venv) $ toil rsync-cluster -z us-west-2a <cluster-name> sort.py :/root
```
This should be the same file as elsewhere in the tutorial.

    (venv) $ toil ssh-cluster -z us-west-2a <cluster-name>

Remember to replace ``<cluster-name>`` with your actual cluster name, and feel
free to use your own cluster configuration and/or workflow files. For more
information on this step, visit :ref:`StaticProvisioning`.

Now that we are inside the cluster, a Kubernetes environment should have already
been configured and running. To verify this, simply run: ::

    $ kubectl get nodes

> You should see a ``master`` node with the ``Ready`` status. Depending on the

Reviewer: ```suggestion
You should see a leader node with the ``Ready`` status. Depending on the
```

Is our Kubernetes version still calling them `master` nodes? I thought Kubernetes switched to "control plane", while Toil mostly says generic "leader" for all schedulers.

Author: Yeah,

```
root@ip-172-31-26-237:/# kubectl get nodes
NAME                                          STATUS   ROLES    AGE   VERSION
ip-172-31-26-237.us-west-2.compute.internal   Ready    master   37h   v1.19.3
ip-172-31-5-230.us-west-2.compute.internal    Ready    <none>   37h   v1.19.3
```

It looks like Kubernetes changed it in [v1.20](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.20.md#urgent-upgrade-notes) but we are using [v1.19.3](https://github.com/DataBiosphere/toil/blob/releases/5.9.x/src/toil/provisioners/abstractProvisioner.py#L840). Changing it to "leader" makes sense though.

number of worker nodes you set to create upfront, you should also see them
displayed here.

Additionally, you can also verify that the metrics server is running: ::

    $ kubectl get --raw "/apis/metrics.k8s.io/v1beta1/nodes"

> If there is a JSON response and you are not seeing any errors, that means the

Reviewer: Maybe there could be an example JSON response? Not everyone knows JSON on sight.

metrics server is set up and running, and you are good to start running workflows.

.. note::
    It'll take a while for all nodes to get set up and running, so you might
    not be able to see all nodes running at first. You can start running
    workflows already, but Toil might complain until the necessary resources
    are set up and running.

Now we can run the workflow: ::

    $ python sort.py \
            --provisioner aws
            --batchSystem kubernetes \
            aws:<region>:<job-store-name>

Make sure to replace ``<region>`` and ``<job-store-name>``. It is **required**

> to use an AWS job store when using the Kubernetes batch system.

Reviewer: ```suggestion
to use a cloud-accessible job store like AWS or Google when using the Kubernetes batch
system.
```
